src/citrine/informatics/generative_design.py

```diff
@@ -128,9 +136,11 @@ def __init__(
         min_fingerprint_similarity: float,
         mutation_per_seed: int,
         structure_exclusions: Optional[List[StructureExclusion]],
```

Comment:
If we need to make other tweaks, let's fix this. Looks like the type-hinting should drop `Optional` and specify a default value of `[]`.

Reply:
@anoto-moniz Mostly because I don't frequently find myself writing highly clean code at the citrine-python standard of quality, and don't often consider the nuances between using something like `Optional` vs. other alternatives, I want to better understand:

Why we wouldn't want to use `Optional` here?  On the pymithril side of this Kaai, Sean, and I came to decide to use `Optional` for both `structure_exclusions` and `min_substructure_counts` arguments (ex: comment thread https://github.com/CitrineInformatics/pymithril/pull/16#discussion_r1182590037)

Reply:
They're correct that you've got to be careful when setting containers as default values, as the article they link details. The thing is that it's only relevant if you're modifying the collection you've been given, which is to say almost never. Hence why I tend to still default towards using them.

Their way is definitely safer, though, in that the issue cannot occur. So you'd set this argument to default to `None`, and then in the method body assign it `[]` if it's `None` and the value otherwise. You can do that easily with something like

`self.structure_exclusions = structure_exclusions if structure_exclusions else []`

Or you can use a slick Python idiom to reduce repetition:

`self.structure_exclusions = structure_exclusions or []`

Reply:
Makes sense, thanks!  I went with the latter suggestion and did the same for `min_substructure_counts`:
`self.min_substructure_counts = min_substructure_counts or {}`).